src/components/TrackListItem.js

```diff
@@ -0,0 +1,113 @@
+import PropTypes from "prop-types";
+import {
+    Container,
+    Row,
+    Col
+  } from "reactstrap";
+import {TrackFilePicker} from './TrackFilePicker';
+import {TrackTypeDropdown} from './TrackTypeDropdown';
+import {TrackDeleteButton} from './TrackDeleteButton';
+import {TrackSettingsButton} from './TrackSettingsButton';
+import React, { useEffect, useReducer, useRef } from 'react';
+
+
+export const TrackListItem = ({
+    trackProps,
+    // trackProps expects an object with
+      // trackType: string
+      // trackFile: file object / undefined
+      // availableTracks: array of tracks(see types.ts)
+      // trackColorSettings: object(aka. colorScheme)
+      // availableColors: array of ColorPalletes
```

Comment:
I think we don't want to have the available options as part of the `trackProps` object; we probably want everything in `trackProps` to be about the track that this controls. Otherwise we're going to end up having to answer questions like, "What if `onChange` is called and it is trying to change the list of available colors?".